libs/ui/src/invalid_card_screen.tsx

```diff
@@ -1,47 +1,75 @@
 import { DippedSmartCardAuth, InsertedSmartCardAuth } from '@votingworks/types';
 
-import { fontSizeTheme } from './themes';
 import { Main } from './main';
 import { Prose } from './prose';
-import { Screen } from './screen';
 import { RotateCardImage } from './rotate_card_image';
+import { Screen } from './screen';
+import { fontSizeTheme } from './themes';
 import { H1, P } from './typography';
 
-type LoggedOutReason =
-  | DippedSmartCardAuth.LoggedOut['reason']
-  | InsertedSmartCardAuth.LoggedOut['reason'];
+type ReasonAndContext = Pick<
+  DippedSmartCardAuth.LoggedOut | InsertedSmartCardAuth.LoggedOut,
+  'reason' | 'cardJurisdiction' | 'cardUserRole' | 'machineJurisdiction'
+>;
 
 export interface Props {
-  reason: LoggedOutReason;
+  reasonAndContext: ReasonAndContext;
   recommendedAction?: string;
 }
 
 export function InvalidCardScreen({
-  reason,
+  reasonAndContext,
   recommendedAction: recommendedActionOverride,
 }: Props): JSX.Element {
+  const { cardJurisdiction, cardUserRole, machineJurisdiction, reason } =
+    reasonAndContext;
+
   let graphic: JSX.Element | null = null;
   let heading = 'Invalid Card';
   let errorDescription = '';
   let recommendedAction =
-    'Please insert a valid Election Manager or System Administrator card.';
-
-  if (reason === 'card_error') {
-    graphic = <RotateCardImage />;
-    // The only cause we currently know of for a card error
-    heading = 'Card is Backwards';
-    recommendedAction = 'Remove the card, turn it around, and insert it again.';
-  }
-
-  if (reason === 'election_manager_wrong_election') {
-    errorDescription =
-      'The inserted Election Manager card is programmed for another election ' +
-      'and cannot be used to unlock this machine.';
-  }
+    recommendedActionOverride ?? 'Please insert a valid card.';
```

Comment:
Switched from "Please insert a valid Election Manager or System Administrator card" to "Please insert a valid card" as the default recommended action since this component is used not only in VxAdmin and VxCentralScan but also in VxScan, where poll worker cards are also valid. (When this component was first created, it was only used in VxAdmin and VxCentralScan.)

In most contexts, we do already provide a `recommendedAction` prop, so this is fairly inconsequential.